docs/en/Community-Articles/2024-05-09-what-is-object-to-object-mapping/post.md

```diff
@@ -0,0 +1,122 @@
+
+            return ObjectMapper.Map<Customer, CustomerDto>(customer);
+        }
+````
+
+# What is AutoMapper?
+
+![Swagger](./images/automapper.png)
+
+Automapper is a .net library that automates object to object mapping. The implementation of Automapper is quite simple. In this chapter I will show you how to use the Automapper library in an ABP-based application. For this reason, I assume that you already have an ABP-based application created. If you have not yet created an ABP-based application, please follow the [Getting Started documentation](https://docs.abp.io/en/abp/latest/Getting-Started-Create-Solution?UI=MVC&DB=EF&Tiered=No).
+
+Create a domain entity similar to this one: 
+
+````csharp
+    public class Customer : FullAuditedAggregateRoot<Guid>
+    {
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime BirthDay { get; set; }
+        public DateTime MembershipDate { get; set; }
+    }
+````
+`Custumer` contains a lot of attributes. we don't want to give all of these attributes to users, so create dto with only the necessary attributes: 
+
+````csharp
+    public class CustomerGetDto
+    {
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime BirthDay { get; set; }
+    }
+````
+
+You can use this code to list customers:
+
+````csharp
+public virtual async Task<PagedResultDto<CustomerGetDto>> GetListAsync(GetCustomersInput input)
+ {
+     var totalCount = await _customerRepository.GetCountAsync(input.FilterText, input.FirstName, input.LastName, input.BirthDayMin, input.BirthDayMax, input.MembershipDateMin, input.MembershipDateMax);
+     var items = await _customerRepository.GetListAsync(input.FilterText, input.FirstName, input.LastName, input.BirthDayMin, input.BirthDayMax, input.MembershipDateMin, input.MembershipDateMax, input.Sorting, input.MaxResultCount, input.SkipCount);
+
+     return new PagedResultDto<CustomerGetDto>
+     {
+         TotalCount = totalCount,
+         Items = ObjectMapper.Map<List<Customer>, List<CustomerGetDto>>(items)
+     };
+ }
+````
+
+Update your `YourApplicationAutoMapperProfile` class so that `Customer` and `CustomerGetDto` entities can convert to each other.
+
+````csharp
+public class YourApplicationAutoMapperProfile : Profile
+{
+    public YourApplicationAutoMapperProfile()
+    {
+        CreateMap<Customer, CustomerGetDto>();
+    }
+}
+````
+The swagger image of the request will be like this. Only the fields in Dtoda will appear.
+
+![Swagger](./images/swagger1.png)
+
+In some scenarios you may want to make some customizations when matching. If you want to use `CustomerGetDto` in this way:
+
+````csharp
+  public class CustomerGetDto
+  {
+      public string? FullName { get; set; }
+      public int Age { get; set; }
+  }
+````
+Autommapper cannot map this automatically, you need to specify the exception. Update your `YourApplicationAutoMapperProfile`.
```

Comment:
```suggestion 
AutoMapper can't map these properties automatically, because they do not exist in the source object, which is the `Customer` entity. Therefore, you need to specify the exception and update your `YourApplicationAutoMapperProfile` class as follows:
```